src/Constructorio_NET.Tests/client/modules/BrowseTests.cs

```diff
@@ -108,6 +108,40 @@ public async Task GetBrowseResultsWithCollection()
             Assert.IsNotNull(res.ResultId, "ResultId should exist");
         }
 
+        [Test]
+        public async Task GetBrowseResultsShouldReturnResultWithHiddenFields()
+        {
+            string requestedHiddenField = "testField";
+            BrowseRequest req = new BrowseRequest(this.FilterName, this.FilterValue)
+            {
+                UserInfo = this.UserInfo,
+                HiddenFields = new List<string> { requestedHiddenField }
+            };
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            BrowseResponse res = await constructorio.Browse.GetBrowseResults(req);
+            var returnedHiddenfield = res.Response.Results[0].Data.Metadata[requestedHiddenField];
+
+            Assert.NotNull(res.ResultId, "Result id exists");
+            Assert.NotNull(returnedHiddenfield, "Hidden field returned");
+        }
+
+        [Test]
+        public async Task GetBrowseResultsShouldReturnResultWithHiddenFacets()
+        {
+            string requestedHiddenFacet = "Brand";
+            BrowseRequest req = new BrowseRequest(this.FilterName, this.FilterValue)
+            {
+                UserInfo = this.UserInfo,
+                HiddenFacets = new List<string> { requestedHiddenFacet }
+            };
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            BrowseResponse res = await constructorio.Browse.GetBrowseResults(req);
+            string returnedHiddenFacet = res.Response.Facets.Find(el => el.Hidden).DisplayName;
+
+            Assert.NotNull(res.ResultId, "Result id exists");
```

Comment:
Should we add a check here to see if the `hidden` attribute on the facet is set to `true`?